core/src/avm2/globals/flash/geom/point.rs

```diff
@@ -91,7 +91,9 @@ pub fn add<'gc>(
 ) -> Result<Value<'gc>, Error> {
     if let Some(mut this) = this {
         if let Some(other) = args.get(0) {
-            let mut other_obj = other.coerce_to_object(activation)?;
+            let mut other_obj = other
+                .as_object()
+                .ok_or("Cannot add point with null or undefined")?;
```

Comment:
Instead of specialized message for every method, maybe let's just make a separate generic method that returns `Result` like `coerce_to_object` (with message like "cannot call methods on non-objects"), but doesn't attempt to create a `PrimitiveObject` for no reason. Then we can rely on Rust stack traces and (upcoming? :) ) AS stack traces for error info